src/mprisbluezproxy.cpp

```diff
@@ -0,0 +1,592 @@
+// -*- c++ -*-
+
+/*!
+ *
+ * Copyright (C) 2018 Jolla Ltd.
+ *
+ * Contact: Jarkko Lehtoranta <jarkko.lehtoranta@jolla.com>
+ * Author: Jarkko Lehtoranta <jarkko.lehtoranta@jolla.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+
+#include "mprisbluezproxy.h"
+
+#include "mprismanager.h"
+
+#include "mpriscontroller.h"
+#include "mprisplayer.h"
+
+#include <qqmlinfo.h>
+
+#include <QDBusConnection>
+#include <QDBusConnectionInterface>
+#include <QDBusMetaType>
+#include <QDBusMessage>
+#include <QDBusReply>
+
+#include <QtCore/QSignalMapper>
+
+static const QString mprisObjectPath = QStringLiteral("/org/mpris/MediaPlayer2");
+static const QString dBusBluezService = QStringLiteral("org.bluez");
+static const QString dBusBluezMediaInterface = QStringLiteral("org.bluez.Media1");
+static const QString dBusBluezMediaRegisterMethod = QStringLiteral("RegisterPlayer");
+static const QString dBusBluezMediaUnregisterMethod = QStringLiteral("UnregisterPlayer");
+
+MprisBluezProxy::MprisBluezProxy(QObject *parent)
+    : QObject(parent)
+    , m_currentController(NULL)
+    , m_bluezRegistered(false)
+{
+    qmlInfo(this) << "Creating MprisBluezProxy";
+
+    // Create mpris bluez player and make connections for it
+    m_bluezPlayer = QSharedPointer<MprisPlayer>(new MprisPlayer(this));
```

Comment:
Is there reason to have this as QSharedPointer when it's also parented? Or actually, is there reason to have it even dynamically allocated?